### DISCUSSION TITLE
Link to a second CSV in a list

### DISCUSSION THREAD

**Discussion by weisgerberm** [cobwebch/external_import#317] [Q&A]
Hi,

thanks for the great extension :) 
I have set everything up and am importing a CSV with a list of product feeds. In these entries there is a link to another CSV with the corresponding products. Now I'm not sure how best to implement this and would like some advice.

---
I am trying to visualize the scenario:
`FeedList.csv is mapped to model "Feed".`
The CSV looks something like this:
```
FeedID, FeedName, ProductsLink
1,test,https://example.org/feed1.csv
2,test,https://example.org/feed2.csv
```
`feed1.csv: is mapped to model "Product"`
```
ProductID,ProductName,...
1,Apple,...
```
I would like to map these entries from the "ProductsLink" field to the "Product" model and reference them via mm-table with feed data record. 

Can the importer handle something like this, that something is reloaded during import? Otherwise I would have tried the following, which is not particularly "nice":
1. import the feeds.
2. once imported, I manually empty the cache and dynamically extend the TCA of the "Product" model based on the feed entries found:
```
 'external' => [
		'general' => [
			x+n => [
				'connector' => 'csv',
				'parameters' => [
//					'filename' => 'https://example.org/feed1.csv',
					'skip_rows' => 1,
					'encoding' => 'utf8'
				],
				....
			]
		],
	],
```
3. And now perform another import so that the children are loaded
But this is not nice, because in this case I create a database connection in my TCA/Overrides/products.php to extend the TCA dynamically.

Surely there is a more elegant solution for this?

**Comment by fsuter** [comment]
Hi. The way I would do this is with a user function that triggers another import, using the API that External Import provides (see https://docs.typo3.org/p/cobweb/external_import/7.2/en-us/Developer/Api/Index.html#import-api). It would rely on a configuration that does not have a connector information and it would be up to your user function to get the data (you can still use the connector service, but have to call it programmatically yourself) and pass the data to the Importer.

I have a similar scenario myself, where I import real estate items, and for each item, I need to call the provided API to retrieve the brokers related to that item. This is done in a user function, which imports the brokers (if needed) and maps them to their primary keys. The user function returns a comma-separated list of uids from the brokers table.

I hope this is helpful.

**Comment by weisgerberm** [comment]
Thanks for the quick reply, I'll try it out later after work. 👍
